data-processing/scripts/run_pipeline.py

```diff
@@ -389,8 +399,10 @@ def run_commands_for_arxiv_ids(
                 continue
             # If no other command filters have been specified and the user didn't
             # request this command, skip it.
-            if args.start is None and args.end is None and args.entities is None:
+            if args.start is None and args.end is None and entities == []:
```

Comment:
I'm pretty sure that what you want here is just

```python
if not args.start and not args.end and not entities:
  ...
```

Consider:

```text
python3
Python 3.8.6 (default, Oct  8 2020, 14:06:32)
[Clang 12.0.0 (clang-1200.0.32.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> if []:
...     print("HEY")
... else:
...     print("NO")
...
NO
>>> a = None
>>> b = None
>>> c = []
>>> d = None
>>> if not a and not b and not c:
...     print("HEY")
... else:
...     print("NO")
...
HEY
>>> if a is None and b is None and not c:
...     print("HEY")
... else:
...     print("NO")
...
HEY
>>> if not a and not b and not d:
...     print("HEY")
... else:
...     print("NO")
...
HEY
>>> if not a and b and not c:
...     print("HEY")
... else:
...     print("NO")
...
NO
>>>
```

Reply:
That said, if you prefer explicitness over what may not invoke 100% confidence in readers then I am totally A-OK with what you have for pretty much all my nits!

Reply:
This works! I'm usually explicit to a fault in these scripts XD. I think your way looks great too; I'll switch it over!